tern/analyze/default/container/image.py

```diff
@@ -23,10 +23,29 @@
 logger = logging.getLogger(constants.logger_name)
 
 
+def download_container_image(image_tag_string):
+    '''Download the docker image and convert it into oci format'''
+    try:
+        # extract the docker image
+        docker_image = image_tag_string
+        if 'docker://' in image_tag_string:
+            docker_image = image_tag_string.split('docker://')[1]
+        image_attr = general.parse_image_string(docker_image)
+        oci_image = 'oci://{0}/{1}'.format(
+            rootfs.working_dir, image_attr.get('name'))
+        docker_image = 'docker://{0}'.format(docker_image)
+        process = subprocess.Popen(['skopeo', 'copy', docker_image, oci_image])
```

Comment:
Any reason why picking `subprocess.Popen` rather than `subprocess.check_output`?

Reply:
`Popen` runs commands in a separate process ( child process) instead of running as part of the same process so It is kind of like a less burden and a little secure I feel to the current process.

Reply:
This is actually an open execution path. `process.wait()` without detecting the end of execution may cause some non-determinism. My suggestion for now is to use `subprocess.check_output` so we are sure the process has ended before we go poking at the downloaded files. It should also resolve the security linting errors.

In the future, why not try `asyncio` or `multiprocessing` modules to reduce tern's execution time? There are plenty of places where we can apply async and parallel processing :)

Reply:
I can surely use `subprocess.check_output` as you suggested but using `process.wait()` let the parent process wait for the child process to finish first. It is a purely deterministic flow here. Maybe I can add a timeout to `process.wait()` so that it makes sure until this timeout the execution should get stopped else raise a Timeout exception.
Anyways, let me use `subprocess.check_output` so that I can resolve security linting issues.